emlib/timer.rs

```diff
@@ -0,0 +1,442 @@
+}
+
+#[inline]
+fn timer3() -> &'static Timer {
+    unsafe { transmute(GET_TIMER3()) }
+}
+
+pub fn capture_get(timer: &Timer, ch: u32) -> u32 {
+    unsafe { STATIC_INLINE_TIMER_CaptureGet(transmute(timer), ch) }
+}
+
+pub fn compare_buf_set(timer: &Timer, ch: u32, val: u32) {
+    unsafe { STATIC_INLINE_TIMER_CompareBufSet(transmute(timer), ch, val) }
+}
+
+pub fn compare_set(timer: &Timer, ch: u32, val: u32) {
+    unsafe { STATIC_INLINE_TIMER_CompareSet(transmute(timer), ch, val) }
+}
+
+pub fn counter_get(timer: &Timer) -> u32 {
+    unsafe { STATIC_INLINE_TIMER_CounterGet(transmute(timer)) }
+}
+
+pub fn counter_set(timer: &Timer, val: u32) {
+    unsafe { STATIC_INLINE_TIMER_CounterSet(transmute(timer), val) }
+}
+
+pub fn enable(timer: &Timer, enable: bool) {
+    unsafe { STATIC_INLINE_TIMER_Enable(transmute(timer), enable) }
+}
+
+pub fn init(timer: &Timer, init: &Init) {
+    unsafe { TIMER_Init(transmute(timer), transmute(init)) }
+}
+
+pub fn init_cc(timer: &Timer, ch: u32, init: InitCC) {
+    unsafe { TIMER_InitCC(transmute(timer), ch, transmute(&init)) }
+}
+
+pub fn init_dti(timer: &Timer, init: InitDTI) {
+    unsafe { TIMER_InitDTI(transmute(timer), transmute(&init)) }
+}
+
+pub fn enable_dti(timer: &Timer, enable: bool) {
+    unsafe { STATIC_INLINE_TIMER_EnableDTI(transmute(timer), enable) }
+}
+
+pub fn get_dti_fault(timer: &Timer) -> u32 {
+    unsafe { STATIC_INLINE_TIMER_GetDTIFault(transmute(timer)) }
+}
+
+pub fn clear_dti_fault(timer: &Timer, flags: u32) {
+    unsafe { STATIC_INLINE_TIMER_ClearDTIFault(transmute(timer), flags) }
+}
+
+pub fn int_clear(timer: &Timer, flags: u32) {
+    unsafe { STATIC_INLINE_TIMER_IntClear(transmute(timer), flags) }
+}
+
+pub fn int_disable(timer: &Timer, flags: u32) {
+    unsafe { STATIC_INLINE_TIMER_IntDisable(transmute(timer), flags) }
+}
+
+pub fn int_enable(timer: &Timer, flags: u32) {
+    unsafe { STATIC_INLINE_TIMER_IntEnable(transmute(timer), flags)}
+}
+
+pub fn int_get(timer: &Timer) -> u32 {
+    unsafe { STATIC_INLINE_TIMER_IntGet(transmute(timer)) }
+}
+
+pub fn int_get_enabled(timer: &Timer) -> u32 {
+    unsafe { STATIC_INLINE_TIMER_IntGetEnabled(transmute(timer)) }
+}
+
+pub fn int_set(timer: &Timer, flags: u32) {
+    unsafe { STATIC_INLINE_TIMER_IntSet(transmute(timer), flags) }
+}
+
+pub fn lock(timer: &Timer) {
+    unsafe { STATIC_INLINE_TIMER_Lock(transmute(timer)) }
+}
+
+pub fn reset(timer: &Timer) {
+    unsafe { TIMER_Reset(transmute(timer)) }
+}
+
+pub fn top_buf_set(timer: &Timer, val: u32) {
+    unsafe { STATIC_INLINE_TIMER_TopBufSet(transmute(timer), val) }
+}
+
+pub fn top_get(timer: &Timer) -> u32 {
+    unsafe { STATIC_INLINE_TIMER_TopGet(transmute(timer)) }
+}
+
+pub fn top_set(timer: &Timer, val: u32) {
+    unsafe { STATIC_INLINE_TIMER_TopSet(transmute(timer), val) }
+}
+
+pub fn unlock(timer: &Timer) {
+    unsafe { STATIC_INLINE_TIMER_Unlock(transmute(timer)) }
+}
+
+
+pub static TIMER_IF_OF: u32     = (0x1 << 0);
```

Comment:
Also, we should be careful where we use `pub static` for identfiers, compared to `pub const`.
See: https://github.com/rust-lang/rfcs/blob/master/text/0246-const-vs-static.md